io-engine/src/bdev/nexus/nexus_child.rs

```diff
@@ -805,6 +808,7 @@ impl<'c> NexusChild<'c> {
         }
 
         self.set_state(ChildState::Faulted(reason));
+        self.faulted_at = Some(Utc::now());
```

Comment:
When the child is "unfaulted", should this be cleared?

Reply:
We discussed that on GTM-464, and agreed that we can retain it as an indication of last faulted timestamp even though child may be currently unfaulted/healthy. Every new fault will overwrite this with latest timestamp.

Reply:
Hmm ok though not sure how we can use it, as when the nexus/process restarts well be losing this information anyway. If we need this sort of information we should persist it to etcd instead, perhaps something to consider in the future should we need to.

Reply:
yeah I can record this as a future task. Moreover, when nexus restarts/moves we are not going to deal with partial rebuilds, and the purpose of adding this field at this time is that this might help us take more informed decisions regarding partial rebuild.

Reply:
In the future we might do partial rebuilds across restarts, and also this information might be useful for thin provisioning as well, so it's not just for partial rebuild.